kafka/cluster_collector.sh

```diff
@@ -0,0 +1,159 @@
+##********************************************************************************************************************
+## The purpose of this tool is to extract kafka & zookeeper related configuration and log files for troubleshooting. 
+## Following are the list of files that are extracted. Please note that not all files exists in an environment. 
+## All properties with the word "password" in it are replaced with "***" 
+#=============================================================#
+# kafka files and the path variables where they are expected
+# BROKER_CONFIG_PATH
+#   server.properties
+# BROKER_LOG_PATH
+#   server.log
+#   kafkaServer.out
+#   kafka-authorizer.log
+#   controller.log
+#   state-change.log
+# BROKER_JAAS_CONFIG
+#   kafka_server_jaas.conf
+# ZOOKEEPER_CONFIG
+#   zookeeper.properties
+#   zoo.cfg
+#   log4j.properties
+# ZOOKEEPER_LOG_PATH
+#   zoo.log
+# ZOOKEEPER_JAAS_CONFIG
+#   zookeeper_jaas.conf
+# ZOOKEEPER_LOG_PATH
+#   zookeeper.out
+# BROKER_BIN_PATH
+#   kafka-topics/.sh
+#   kafka-topics/.sh
+#   kafka-broker-api-versions/.sh
+#   kafka-consumer-groups/.sh 
+#=============================================================#      
+##
+## In addition to the files above the script also extract the following OS related information - 
+## 1. file system & directory size 
+## 2. io stats
+## 3. file descriptors
+## 4. cpu & memory 
+## 5. contents of the hosts file 
+## 6. output of kafka-topics.sh topic describe 
+## 
+##********************************************************************************************************************
+##********************************************************************************************************************
+## Last Modification Date : 10/29/2021
+## Description		 : Script functionality enhanced to add information related to iostat, df, file descriptor
+##                          cpu & memory info 
+##********************************************************************************************************************
+##********************************************************************************************************************
+
+clear
+
+#GLOBAL VARIABLES
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+INFO_DIR=/tmp/InstaCollection_$(date +%Y%m%d%H%M)
+
+#Collect environment info (VM/docker)
+read -p "Enter your kafka environment (vm/docker) :" kenv
+
+if [[ "${kenv}" == "vm" ]]; then
+  #Collect user info.
+  read -p "Enter username for login on Kafka cluster nodes (Press Enter for default admin) :" user
+  [ -z "${user}" ] && user='admin'
+
+  #user='rahulchakrabarty'
+
+  read -p "Enter Identity file path:" id_file
+  if [[ ! -f ${id_file} || ! -s ${id_file} ]]; then
+      echo "$id_file File not found!" 
+      exit 1
+  fi
+
+  #id_file='/Users/rahulchakrabarty-instaclustr/.ssh/rahulchakrabarty-instaclustr'
+elif [[ "${kenv}" == "docker" ]]; then
+  read -p "Enter docker home directory :" docker_home
```

Comment:
It's not clear to me as a user what is this directory supposed to be? It looks like it's used further down as a temporary location to copy the node_collector.sh script to and run from. You should specify that here - otherwise people are going to assume you are asking for a directory on the host machine. Also, do we actually need the user to specify it? Can we just use /tmp, ~, something else?